### DISCUSSION TITLE
Manifests - Required Files

### DISCUSSION THREAD

**Discussion by ljtill** [microsoft/winget-pkgs#37114] [Q&A] [answered]
Within a given manifest folder, what are the required files for each manifest version? 
After looking through several examples, it seems that typically only the `package_name.yaml` is required but can vary?

However, if we use Visual Studio manifests as an example there are two different approaches (Stable / Preview)...

[Visual Studio 2022 Enterprise 17.0.1](https://github.com/microsoft/winget-pkgs/tree/master/manifests/m/Microsoft/VisualStudio/2022/Enterprise/17.0.1)
- _Microsoft.VisualStudio.2022.Enterprise.yaml_

[Visual Studio 2022 Enterprise Preview 17.0.0 Preview 2.1](https://github.com/microsoft/winget-pkgs/tree/master/manifests/m/Microsoft/VisualStudio/2022/Enterprise-Preview/17.0.0%20Preview%202.1)
- _Microsoft.VisualStudio.2022.Enterprise-Preview.installer.yaml_
- _Microsoft.VisualStudio.2022.Enterprise-Preview.locale.en-US.yaml_
- _Microsoft.VisualStudio.2022.Enterprise-Preview.installer.yaml_

**Comment by OfficialEsco** [comment, marked as answer]
[Visual Studio 2022 Enterprise 17.0.1](https://github.com/microsoft/winget-pkgs/tree/master/manifests/m/Microsoft/VisualStudio/2022/Enterprise/17.0.1) is a singleton ManifestType. It can contain 1 installer and 1 Locale.

[Visual Studio 2022 Enterprise Preview 17.0.0 Preview 2.1](https://github.com/microsoft/winget-pkgs/tree/master/manifests/m/Microsoft/VisualStudio/2022/Enterprise-Preview/17.0.0%20Preview%202.1) is a Multi Manifest, This splits the singleton into Version, Installer and defaultLocale manifests.
The Version ManifestType only contains the `PackageIdentifier`, `PackageVersion`, `DefaultLocale`
The Installer ManifestType can a contain bunch of installers, which means multiple languages and user and machine scope switches
The defaultLocale ManifestType contains the default metadata, should usually be in English (en-US)
The locale ManifestType can contains any translation or metadata for another language.

Example: [Mozilla Firefox](https://github.com/microsoft/winget-pkgs/tree/master/manifests/m/Mozilla/Firefox/95.0)

And yes we prefer Multi Manifests, any new versions/packages are converted to a Multi Manifest, the singletons are from ManifestVersion 0.0.1

**Reply by ljtill** [reply]
Thanks for the clarification @OfficialEsco
